src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
I agree with @AlexGhiondea that TickCount is good enough as a global seed.

Although it may become more common to start a bunch of processes for scenarios for which AppDomains would have previously been used, relying on `System.Random` for important logic is discouraged anyway and applications doing that would have already faced seeding problems during startup.

Not to forget that if an applications requires "good" seeds, they most probably also need "good" random numbers so it should be using `RNGCryptoServiceProvider` instead of `System.Random`. That would be best addressed through documentation.